File: portrefs/ion/ui/ISettings.java
package ion.ui;

import arc.*;
import arc.util.*;
import arc.math.*;
import mindustry.*;
import mindustry.gen.*;

import static arc.Core.*;
import static mindustry.Vars.*;

public class ISettings{
    private static float waitLength;
    
    public static void load(){
        ui.settings.addCategory("Ion", stable -> {
            stable.textPref("Input", "...", s1 -> {
                switch(s1){
                    case "horny":
                        app.exit();
                    break;
                    
                    case "amogus":
                        ui.showInfo("sugosmu garitesu");
                    break;
                    
                    case "seq":
                        waitLength = Mathf.random(600f);
                        ui.loadfrag.show("seqs uwu\nwait " + (waitLength / 60f) + " seconds...");
                        Time.runTask(waitLength, () -> ui.loadfrag.hide());
                    break;
                    
                    case "screw you 192.168.1.34":
                        Sounds.wind3.play(Mathf.random(200f));
                    break;
                    
                    case "how do i pronounce mindustry":
                        ui.showInfo(tree.get("texts/methods.txt").readString());
                    break;
                    
                    case "plus one":
                        Log.warn("angry cheesy-chan noises");
                    break;
                    
                    case "tantos":
                        Threads.throwAppException(new RuntimeException(tree.get("texts/tantos.txt").readString()));
                    break;
                    
                    case "pet the cheesy-chan":
                        ui.showConfirm("Question", "have you gotten [accent]consent[]?", () -> ui.showInfo("the cheesy-chan has been pet."));
                    break;
                }
            });
        });
    }
}

File: src/ion/ui/ISettings.kt
package ion.ui

import arc.Core
import arc.Core.app
import arc.util.*
import arc.math.Mathf
import mindustry.Vars.*
import mindustry.gen.*

import ion.content.*

object ISettings{
    fun load(){
        ui.settings.addCategory("Ion: Global", Icon.right){
            
            it.checkPref("Effect Reduction", false){
                Core.settings.put("effectreduction", it)
            }
            
            it.textPref("Input", "..."){
                when(it){
                    "horny" -> app.exit()
                    
                    "amogus" -> ui.showInfo("sugosmu garitesu")
                    
                    "seq" -> {
                        var waitLength = Mathf.random(600f)
                        ui.loadfrag.show("""
                            seqs uwu
                            wait ${waitLength / 60f} seconds...
                        """.trimIndent())
                        Time.runTask(waitLength) { ui.loadfrag.hide() }
                    }
                    
                    "screw you 192.168.1.34" -> Sounds.wind3.play(Mathf.random(200f))
                    
                    "how do i pronounce mindustry" -> ui.showInfo(tree["texts/methods.txt"].readString())
                    
                    "plus one" -> Log.warn("angry cheesy-chan noises")
                    
                    "tantos" -> Threads.throwAppException(RuntimeException(tree["texts/tantos.txt"].readString()))
                    
                    "pet the cheesy-chan" -> ui.showConfirm("Question","have you gotten [accent]consent[]?") { ui.showInfo("the cheesy-chan has been pet.") }
                }
            }
            
        }
        
        ui.settings.addCategory("Ion: Units", Icon.right){
            
            it.sliderPref("Xeus Line Effect Count", 10, 0, 100, 1){
                Core.settings.put("xeuslinecount", it)
                
                "$it"
            }
            
            it.sliderPref("Xeus Line Effect Slowness", 10, 1, 100, 1){
                Core.settings.put("xeuslineeffectslowness", it.toFloat())
                
                "$it"
            }
        }
    }
}
